### PR TITLE
[tool] Check Java integration test configuration

### PR DESCRIPTION
**Pull request by stuartmorgan**
Rather than ignoring files by filename convention (*ActivityTest.java), look for the @DartIntegrationTest annotation that the test filter actually uses, to prevent accidentally excluding files.

Also validate that the annotation is present when it should be, so that forgetting to add it doesn't cause unexplained hangs when running locally.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: script/tool/lib/src/native_test_command.dart [hunk 1]
```diff
@@ -247,10 +271,17 @@ this command.
     bool ranAnyTests = false;
     bool failed = false;
     bool hasMissingBuild = false;
+    bool hasMisconfiguredIntegrationTest = false;
+    // Iterate through all examples (in the rare case that there is more than
+    // on example); running any tests found for each one. Requirements on what
```
Comment (tarrinneal) [inline script/tool/lib/src/native_test_command.dart:276]: nit: one

File: script/tool/lib/src/native_test_command.dart [hunk 2]
```diff
@@ -622,3 +674,15 @@ class _PlatformResult {
...
+
+/// The state of a .java test file.
+class _JavaTestInfo {
+  const _JavaTestInfo(
+      {required this.usesFlutterTestRunner,
+      required this.hasDartIntegrationTestAnnotation});
+
+  /// Whether the test class uses the FlutterTestRunner runner.
+  final bool usesFlutterTestRunner;
+  //// Whether the test class has the @DartIntegrationTest annotation.
```
Comment (tarrinneal) [inline script/tool/lib/src/native_test_command.dart:686]: A little too commented for my liking.

File: script/tool/lib/src/native_test_command.dart [hunk 3]
```diff
@@ -622,3 +674,15 @@ class _PlatformResult {
...
   final String? error;
 }
+
+/// The state of a .java test file.
+class _JavaTestInfo {
+  const _JavaTestInfo(
+      {required this.usesFlutterTestRunner,
+      required this.hasDartIntegrationTestAnnotation});
+
+  /// Whether the test class uses the FlutterTestRunner runner.
```
Comment (tarrinneal) [inline script/tool/lib/src/native_test_command.dart:684]: optional: runner seems a little redundant at the end of this sentence.

**Review by tarrinneal — APPROVED** [review]
Just a few nits in the comments. Looks good!

**Review by reidbaker — APPROVED** [review]
Approve % comments from @tarrinneal

File: script/tool/lib/src/native_test_command.dart (hunk 3 above)
Comment (stuartmorgan) [reply script/tool/lib/src/native_test_command.dart]: Fixed.

File: script/tool/lib/src/native_test_command.dart (hunk 2 above)
Comment (stuartmorgan) [reply script/tool/lib/src/native_test_command.dart]: Heh, I was about to defend the comment as being marginally useful even though it's not much more information than the name has, until I realized that you meant it had an extra `/`. Good eye; fixed!

File: script/tool/lib/src/native_test_command.dart (hunk 1 above)
Comment (stuartmorgan) [reply script/tool/lib/src/native_test_command.dart]: Fixed.

**Comment by stuartmorgan** [comment]
Overriding stale tree status.

**Pull request merged**